src/noit_check_log.c

```diff
@@ -548,35 +597,51 @@ noit_check_log_bundle_fb_serialize(mtev_log_stream_t ls, noit_check_t *check) {
   uuid_str[0] = '\0';
   mtev_uuid_unparse_lower(check->checkid, uuid_str);
 
-  metrics = noit_check_stats_metrics(c);
+  metrics = in_metrics ? in_metrics : noit_check_stats_metrics(c);
 
   int account_id = account_id_from_name(check_name);
-  void *B = noit_fb_start_metricbatch((SECPART(whence) * 1000) + MSECPART(whence), uuid_str, check_name, account_id);
+  size_t fb_size;
+  int current_in_batch = 0;
+  char *outbuf = NULL;
+  void *buffer = NULL;
+  unsigned int outsize;
 
+  void *B = NULL;
+  uint64_t whence_ms = (SECPART(whence) * 1000) + MSECPART(whence);
+  
   while(mtev_hash_next(metrics, &iter, &key, &klen, &vm)) {
     /* If we apply the filter set and it returns false, we don't log */
     metric_t *m = (metric_t *)vm;
     if(!noit_apply_filterset(check->filterset, check, m)) continue;
     if(m->logged) continue;
+
+    if(!B) B = noit_fb_start_metricbatch(whence_ms, uuid_str, check_name, account_id);
+    current_in_batch++;
     noit_fb_add_metric_to_metricbatch(B, m, 0);
-  }
 
-  size_t fb_size;
-  void *buffer = noit_fb_finalize_metricbatch(B, &fb_size);
-  char *outbuf;
-  unsigned int outsize;
-  noit_check_log_bundle_compress_b64(NOIT_COMPRESS_LZ4, buffer, fb_size, &outbuf, &outsize);
+    if(current_in_batch >= metrics_per_bundle) {
+do_batch:
+      buffer = noit_fb_finalize_metricbatch(B, &fb_size);
+      noit_check_log_bundle_compress_b64(NOIT_COMPRESS_LZ4, buffer, fb_size, &outbuf, &outsize);
```

Comment:
outbuf can come back NULL when this method returns -1, in which case the mtevL a couple lines down will segfault